src/cryptography/hazmat/bindings/_rust/x509.pyi

```diff
@@ -1,15 +1,55 @@
-from cryptography.x509 import (
-    ExtensionType,
-    PrecertificateSignedCertificateTimestamps,
-)
+import datetime
+import typing
 
-def parse_x509_extension(der_oid: bytes, ext_data: bytes) -> ExtensionType: ...
+from cryptography import x509
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.x509.base import PUBLIC_KEY_TYPES
+
+def load_pem_x509_certificate(data: bytes) -> Certificate: ...
+def load_der_x509_certificate(data: bytes) -> Certificate: ...
+def parse_x509_extension(
+    der_oid: bytes, ext_data: bytes
+) -> x509.ExtensionType: ...
 def parse_crl_entry_extension(
     der_oid: bytes, ext_data: bytes
-) -> ExtensionType: ...
-def parse_crl_extension(der_oid: bytes, ext_data: bytes) -> ExtensionType: ...
+) -> x509.ExtensionType: ...
+def parse_crl_extension(
+    der_oid: bytes, ext_data: bytes
+) -> x509.ExtensionType: ...
 def encode_precertificate_signed_certificate_timestamps(
-    extension: PrecertificateSignedCertificateTimestamps,
+    extension: x509.PrecertificateSignedCertificateTimestamps,
 ) -> bytes: ...
 
 class Sct: ...
+
+class Certificate(x509.Certificate):
```

Comment:
Err, why is this required? Just define the functions as returning `x509.Certificate`